src/v/security/gssapi_principal_mapper.cc

```diff
@@ -0,0 +1,482 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "security/gssapi_principal_mapper.h"
+
+#include "security/logger.h"
+#include "vassert.h"
+#include "vlog.h"
+
+#include <boost/algorithm/string/case_conv.hpp>
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+
+#include <charconv>
+
+namespace security {
+static constexpr std::string_view gssapi_name_pattern
+  = R"(([^/@]*)(/([^/@]*))?@([^/@]*))";
+static constexpr std::string_view non_simple_pattern = R"([/@])";
+static constexpr std::string_view parameter_pattern = R"(([^$]*)(\$(\d*))?)";
+static constexpr std::string_view rule_pattern
+  = R"(((DEFAULT)|((RULE:\[(\d*):([^\]]*)](\(([^)]*)\))?(s\/([^\/]*)\/([^\/]*)\/(g)?)?\/?(L|U)?))))";
+
+namespace detail {
+std::vector<gssapi_rule> parse_rules(
+  std::string_view default_realm,
+  std::optional<std::vector<ss::sstring>> unparsed_rules) {
+    static thread_local const re2::RE2 rule_parser(
+      rule_pattern, re2::RE2::Quiet);
+
+    vassert(
+      rule_parser.ok(),
+      "Failed to build rule pattern regex: {}",
+      rule_parser.error());
+
+    if (!unparsed_rules || unparsed_rules.value().empty()) {
+        return {gssapi_rule(default_realm)};
+    }
+
+    std::vector<gssapi_rule> rv;
+    re2::StringPiece default_;
```

Comment:
random thought, i wonder if stringpiece is compatible w/ string_view

Reply:
our internal version of re2 is, but not the one that ships in the latest fedora.

Reply:
Yeah this was the source of my confusion/heartburn earlier today.